x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/result_settings_table/result_settings_table.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+
+import { useValues } from 'kea';
+
+import { EuiTable } from '@elastic/eui';
+
+import { ResultSettingsLogic } from '..';
+
+import { ResultSettingsDisabledFieldsBody } from './result_settings_disabled_fields_body';
+import { ResultSettingsDisabledFieldsHeader } from './result_settings_disabled_fields_header';
+import { ResultSettingsNonTextFieldsBody } from './result_settings_non_text_fields_body';
+import { ResultSettingsNonTextFieldsHeader } from './result_settings_non_text_fields_header';
+import { ResultSettingsTextFieldsBody } from './result_settings_text_fields_body';
+import { ResultSettingsTextFieldsHeader } from './result_settings_text_fields_header';
```

Comment:
SUPER love this being split up into such atomic components - made reading the code super easy!!

One thought - I think it's clear here that these are all Result Settings specific components by them being nested in `/result_settings_table/`, and since none of these files are exported to as top-level components, I think we could lose the prefixing to reduce verbosity:

```js
import { DisabledFieldsHeader } from './disabled_fields_header';
import { DisabledFieldsHeader } from './disabled_fields_header';
import { NonTextFieldsBody } from './non_text_fields_body';
// etc
```

Totally up to you though, just a thought!

Reply:
Oh yeah, I'll definitely do this. Great suggestion.

Reply:
Much cleaner: edaa670ff323404086ad513e1e795c1c846e4e22